bootstore/src/schemes/v0/state_initial_member.rs

```diff
@@ -0,0 +1,411 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+//! FSM API for `State::InitialMember`
+
+use std::collections::BTreeMap;
+
+use crate::schemes::v0::fsm_output::{ApiError, ApiOutput};
+use crate::trust_quorum::{LearnedSharePkgV0, RackSecret, SharePkgV0};
+
+use super::fsm::StateHandler;
+use super::fsm_output::Output;
+use super::messages::{Envelope, Error, RequestType, Response, ResponseType};
+use super::state::{
+    FsmCommonData, RackInitState, RequestMetadata, ShareIdx, State,
+};
+use secrecy::{ExposeSecret, Secret};
+use sled_hardware::Baseboard;
+use uuid::Uuid;
+
+#[derive(Debug)]
+pub struct InitialMemberState {
+    pub pkg: SharePkgV0,
+
+    /// Shares given to other sleds. We mark them as used so that we don't
+    /// hand them out twice. If the same sled asks us for a share, because
+    /// it crashes or there is a network blip, we will return the same
+    /// share each time.
+    ///
+    /// Note that this is a fairly optimistic strategy as the requesting
+    /// sled can always go ask another sled after a network blip. However,
+    /// this guarantees that a single sled never hands out more than one of
+    /// its shares to any given sled.
+    ///
+    /// We can't do much better than this without some sort of centralized
+    /// distributor which is part of the reconfiguration mechanism in later
+    /// versions of the trust quourum protocol.
+    pub distributed_shares: BTreeMap<Baseboard, ShareIdx>,
```

Comment:
Per [this comment](https://github.com/oxidecomputer/omicron/pull/3385/files#r1248290031), I wonder if we should remove `encrypted_shares` from `SharePkgV0` and instead put them here in `InitialMemberState`.

I'm also not convinced by `ShareIdx`, versus popping items from the `encrypted_shares: Vec<...>` and storing those encrypted values in this `BTreeMap` directly.

Reply:
Oh wait, `encrypted_shares` is a single blob, not a `Vec<Vec<u8>>`, so disregard the second part.